src/pal/pal_linux.h

```diff
@@ -85,6 +86,108 @@ namespace snmalloc
         madvise(p, size, MADV_FREE);
       }
     }
+
+    static uint64_t get_entropy64()
+    {
+      // TODO: If the system call fails then the POSIX PAL calls libc
+      // functions that can require malloc, which may result in deadlock.
+
+      // SYS_getrandom API stablized since 3.17.
+      // This fallback implementation is to aid some environments
+      // where SYS_getrandom is provided in kernel but the libc
+      // is not providing getentropy interface.
+
+      union
+      {
+        uint64_t result;
+        char buffer[sizeof(uint64_t)];
+      };
+      ssize_t ret;
+
+      // give a try to SYS_getrandom
+#  ifdef SYS_getrandom
+      static std::atomic_bool syscall_not_working = false;
+      // Relaxed ordering should be fine here. This function will be called
+      // during early initialisation, which will examine the availability in a
+      // protected routine.
+      if (false == syscall_not_working.load(std::memory_order_relaxed))
+      {
+        auto current = std::begin(buffer);
+        auto target = std::end(buffer);
+        while (auto length = target - current)
+        {
+          // reading data via syscall from system entropy pool.
+          // According to both MUSL and GLIBC implementation, getentropy uses
+          // /dev/random (blocking API) as the pool which is indicated by the
```

Comment:
This should also be urandom.  See [myths about urandom](https://www.2uo.de/myths-about-urandom/)